deploy/0-crd.yaml

```diff
@@ -33,13 +33,43 @@ spec:
                 - type: integer
                   multipleOf: 8
                   minimum: 8
+              slotsPerWorker:
+                title: The number of slots per worker used in hostfile
+                description: Defaults to the number of processing units per worker
+                type: integer
+                minimum: 1
             required:
             - gpus
+          - properties:
+              processingUnits:
```

Comment:
Should you also add a check for processingResourceType?

Reply:
That's part of `main.go` as a flag since I thought it was more related to the cluster's spec where controller gets deployed along with `processingUnitsPerNode`. The check is done as part of `convertProcessingResourceType()`. Or maybe this is more `MPIJob` user specific (if they want to specify custom resource) and I should move it to `MPIJobSpec`?

Reply:
Oh right. Based on the experience with the `gpusPerNode` flag, it seems easier for users to understand if we put the options in the CRD. Maybe take them out of the operator flag and just use the job spec?

Reply:
I’ll keep those flags for now for backwards compatibility and then add those fields in MPIJobSpec that overwrites the flags when specified. What do you think?

Reply:
OK, sounds good.